### PR TITLE
HTCONDOR-2400 file xfer missing plugin errors

### PR DESCRIPTION
**Pull request by GregThain**
<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [x] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [x] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [x] Check for version history, if needed
- [x] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/condor_utils/file_transfer.cpp [hunk 1]
```diff
@@ -3497,6 +3505,9 @@ FileTransfer::DoDownload( filesize_t *total_bytes_ptr, ReliSock *s)
 				}
 				try_again = false;
```
Comment (Todd-L-Miller) [inline src/condor_utils/file_transfer.cpp:3506]: This looks silly but probably isn't worth changing to something like
```
try_again = result == TransferPluginResult::ExecFailed;
````

File: src/condor_utils/file_transfer.cpp [hunk 2]
```diff
@@ -3497,6 +3505,9 @@ FileTransfer::DoDownload( filesize_t *total_bytes_ptr, ReliSock *s)
...
 				formatstr(error_buf, "%s", errstack.getFullText().c_str());
+				if (result == TransferPluginResult::ExecFailed) {
+					try_again = true; // not the job's fault
+				}
```
Comment (Todd-L-Miller) [inline src/condor_utils/file_transfer.cpp:3510]: This looks silly but probably isn't worth checking a change to something like the following.
```suggestion
				}
				formatstr(error_buf, "%s", errstack.getFullText().c_str());
				try_again = (result == TransferPluginResult::ExecFailed;
```

**Review by Todd-L-Miller — APPROVED** [review]
Looks good to me.

**Comment by Todd-L-Miller** [comment]
It seems like this wouldn't be too hard to write a test for?

File: src/condor_utils/file_transfer.cpp (hunk 2 above)
Comment (GregThain) [reply src/condor_utils/file_transfer.cpp:3510]: FWIW, my thought was that something else is going to in this block in the future.

**Pull request merged**